Source/Core/DolphinWX/FrameTools.cpp

```diff
@@ -292,6 +292,70 @@ void CFrame::OpenGeneralConfiguration(wxWindowID tab_id)
   m_main_config_dialog->SetFocus();
 }
 
+void CFrame::CycleProfile(bool forward) {
+  //load the hotkey config
+  InputConfig* hk_conf = HotkeyManagerEmu::GetConfig();
+  auto hotkey_controller = hk_conf->GetController(0);
+
+  //load the wiimote config
+  InputConfig* wm_conf = Wiimote::GetConfig();
+  std::string pname(File::GetUserPath(D_CONFIG_IDX));
+
+  //find all profiles
+  pname += "Profiles/" + wm_conf->GetProfileName();
+  std::vector<std::string> sv = Common::DoFileSearch({ pname }, { ".ini" });
+
+  //remove all profiles that do not fit the filter string
+  if (!hotkey_controller->profile_filter.empty()){
+    sv.erase(
+      std::remove_if(
+        sv.begin(), sv.end(), [filter = hotkey_controller->profile_filter](std::string s)
+        {
+          return s.find(filter) == std::string::npos;
+        }
+      ), sv.end()
+    );
+  }
+
+  //check if there are any profiles to cycle through
+  if (sv.empty())
+  {
+    if (hotkey_controller->profile_filter.empty())
+      Core::DisplayMessage("No Profiles found", 3000);
+    else
+      Core::DisplayMessage("No Profiles found matching filter: " + hotkey_controller->profile_filter, 3000);
+
+    return;
+  }
+
+  size_t controllerCount = wm_conf->GetControllerCount();
+
+  //increment/decrement the profile index, clamped to the filtered set of profiles
+  hotkey_controller->profile_index += forward ? 1 : -1;
+  hotkey_controller->profile_index = hotkey_controller->profile_index >= sv.size() ? 0 : hotkey_controller->profile_index;
```

Comment:
This and the following line are doing both sign conversion and truncation without an explicit cast.

Reply:
Fixed.